### PR TITLE
Adding option to overwrite while doing correct_motion and saving to a folder

### PR DESCRIPTION
**Pull request by yger**
Currently, when doing motion correction, if saving to a folder we can not overwrite and an error is launched only at then end, after motion correction has been made. This PR adds the option to overwrite the motion folder

### CODE REVIEW

File: src/spikeinterface/preprocessing/motion.py [hunk 1]
```diff
@@ -316,6 +319,14 @@ def correct_motion(
 
     if folder is not None:
         folder = Path(folder)
+        if overwrite:
+            if folder.exists():
```
Comment (alejoe91) [inline src/spikeinterface/preprocessing/motion.py:323]:
```suggestion
            if folder.is_dir():
```

File: src/spikeinterface/preprocessing/motion.py [hunk 2]
```diff
@@ -316,6 +319,14 @@ def correct_motion(
...
+                import shutil
+
+                shutil.rmtree(folder)
+        else:
+            assert not folder.exists(), f"Folder {folder} already exists"
```
Comment (alejoe91) [inline src/spikeinterface/preprocessing/motion.py:328]:
```suggestion
            assert not folder.is_dir(), f"Folder {folder} already exists"
```

File: src/spikeinterface/preprocessing/motion.py (hunk 1 above)
Comment (alejoe91) [reply src/spikeinterface/preprocessing/motion.py:323]: otherwise `shutil.rmtree` will fail

**Review by alejoe91 — APPROVED** [review]

File: src/spikeinterface/preprocessing/motion.py [hunk 3]
```diff
@@ -204,6 +204,7 @@ def correct_motion(
     recording,
     preset="nonrigid_accurate",
     folder=None,
+    overwrite=False,
```
Comment (samuelgarcia) [inline src/spikeinterface/preprocessing/motion.py:207]: This is not consistent with some otherplace in spikeinterface.

**Comment by alejoe91** [comment]
@yger why did you remove the code that does the overwriting?

**Comment by yger** [comment]
aie, mistake then, i'll check

**Comment by yger** [comment]
I've mixed up some git history I guess, so mayeb the best is to delete this PR and start a new one? I don't know how to properly revert, there are way too many changes in my commit that are not mine here

**Review by alejoe91 — DISMISSED** [review]

**Comment by alejoe91** [comment]
the diff in files look really weird!

@yger I would suggest making a clean PR without messing up with git history ;)

**Comment by yger** [comment]
Now this has been fixed and the diff is legit. Just squash everything

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**